lib/demos/cupertino/cupertino_segmented_control_demo.dart

```diff
@@ -26,19 +24,21 @@ class _CupertinoSegmentedControlDemoState
   String get restorationId => 'cupertino_segmented_control';
 
   @override
-  void restoreState(RestorationBucket oldBucket, bool initialRestore) {
+  void restoreState(RestorationBucket? oldBucket, bool initialRestore) {
     registerForRestoration(currentSegment, 'current_segment');
   }
 
-  void onValueChanged(int newValue) {
+  void onValueChanged(int? newValue) {
     setState(() {
-      currentSegment.value = newValue;
+      if (newValue != null) {
```

Comment:
@guidezpl should I directly imply that this value won't be null or maybe use RestorableIntN ?

Reply:
No null is fine, since an option is always selected